src/groups/mqb/mqba/mqba_configprovider.cpp

```diff
@@ -71,39 +71,22 @@ bool ConfigProvider::cacheLookup(mqbconfm::Response*      response,
     }
 
     // Cache entry is still 'alive'
-    *response = it->second.d_data;  // Assign a copy of the object
+    *config = it->second.d_data;  // Assign a copy of the object
     return true;
 }
 
-void ConfigProvider::onDomainConfigResponseCb(
-    const mqbconfm::Response response,
-    const GetDomainConfigCb& callback)
+void ConfigProvider::cacheAdd(const bslstl::StringRef& key,
+                              const bsl::string&       config)
 {
-    if (response.isFailureValue()) {
-        callback(response.failure().code(), response.failure().message());
-        return;  // RETURN
-    }
-
-    BSLS_ASSERT_OPT(response.isDomainConfigValue());
-
-    BALL_LOG_INFO << "Received domain config for domain '"
-                  << response.domainConfig().domainName() << "': '"
-                  << response.domainConfig().config() << "'";
-    {
-        bslmt::LockGuard<bslmt::Mutex> guard(&d_mutex);  // mutex LOCKED
-
-        // Save to cache
-        CacheEntry cacheEntry;
-        cacheEntry.d_data = response;
-        cacheEntry.d_expireTime =
-            bmqsys::Time::nowMonotonicClock() +
-            bsls::TimeInterval(
-                mqbcfg::BrokerConfig::get().bmqconfConfig().cacheTTLSeconds());
-        d_cache[response.domainConfig().domainName()] = cacheEntry;
-    }
+    BSLMT_MUTEXASSERT_IS_LOCKED_SAFE(&d_mutex);  // mutex LOCKED
 
-    // Call callback
-    callback(0, response.domainConfig().config());
+    CacheEntry cacheEntry;
+    cacheEntry.d_data = config;
+    cacheEntry.d_expireTime =
+        bmqsys::Time::nowMonotonicClock() +
+        bsls::TimeInterval(
+            mqbcfg::BrokerConfig::get().bmqconfConfig().cacheTTLSeconds());
+    d_cache[key] = cacheEntry;
```

Comment:
`d_cache` is a `unordered_map<string,...>`. Inserting a `string` constructed from `string_view` might use a default allocator depending on trait `propagate_on_container_copy_assignment`

Something like this should solve this
`d_cache.insert(bsl::make_pair(bsl::string(key, d_allocator_p), cacheEntry));`